### PR TITLE
Changing the default programming model of Node and Python

### PR DESCRIPTION
**Pull request by khkh-ms**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by khkh-ms** [comment]
Expecting some tests will fail. I will fix them once the pipeline finishes the first build.

File: src/Azure.Functions.Cli/Helpers/NodeJSHelpers.cs [hunk 1]
```diff
@@ -7,28 +7,10 @@ namespace Azure.Functions.Cli.Helpers
...
     {
-        public static void PrintV4ReferenceMessage()
-        {
-            ColoredConsole.Write(AdditionalInfoColor("The new Node.js programming model is in public preview. Learn more at "));
-            PrintNodeV4WikiLink();
-        }
-
-        public static void PrintV4AwarenessMessage()
-        {
-            ColoredConsole.Write(AdditionalInfoColor("Did you know? There is a new Node.js programming model in public preview. Learn how you can try it out today at "));
```
Comment (ejizba) [inline src/Azure.Functions.Cli/Helpers/NodeJSHelpers.cs:18]: I suggest keeping the message and changing text to:

> Did you know? The new Node.js programming model is generally available. Learn how you can try it out today at

File: src/Azure.Functions.Cli/Helpers/NodeJSHelpers.cs [hunk 2]
```diff
@@ -7,28 +7,10 @@ namespace Azure.Functions.Cli.Helpers
 {
     public static class NodeJSHelpers
     {
-        public static void PrintV4ReferenceMessage()
-        {
-            ColoredConsole.Write(AdditionalInfoColor("The new Node.js programming model is in public preview. Learn more at "));
```
Comment (ejizba) [inline src/Azure.Functions.Cli/Helpers/NodeJSHelpers.cs:12]: I suggest keeping the message and changing text to:

> The new Node.js programming model is generally available. Learn more at

**Review by ejizba — COMMENTED** [review]
We still want these messages, we just don't want them saying "Preview" anymore. I especially want the "Awareness" message because that one encourages people using the old model to use the new one instead.

cc @MadhuraBharadwaj-MSFT

File: azure-pipelines.yml [hunk 3]
```diff
@@ -58,7 +58,7 @@ jobs:
     displayName: Set up environment variables
   - task: NodeTool@0
     inputs:
-      versionSpec: '14.x'
+      versionSpec: '18.x'
```
Comment (khkh-ms) [inline azure-pipelines.yml:61]: Node 18.x is required for testing node v4.

**Review by ejizba — APPROVED** [review]
Looks good, but as discussed offline let's do this for 4.25 not 4.24

**Pull request merged**